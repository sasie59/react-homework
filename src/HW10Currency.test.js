import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW10Currency';

test('should HW10Currency', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();

});

test('should  1 usd change 30 twd', () => {
  const { container } = render(<HW />);
  const usdInput = container.querySelector('input[name="usd"]');

  userEvent.type(usdInput, '1');
  expect(container).toMatchSnapshot();
});

test('should 30 twd change 1 usd', () => {
  const { container } = render(<HW />);
  const twdInput = container.querySelector('input[name="twd"]');

  userEvent.type(twdInput, '30');
  expect(container).toMatchSnapshot();
});


