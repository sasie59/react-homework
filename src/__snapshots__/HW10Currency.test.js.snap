// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  1 usd change 30 twd 1`] = `
<div>
  <div>
    <h1>
      HW10Currency
    </h1>
    <h1>
      貨幣轉換
    </h1>
    TWD
    <input
      name="twd"
      type="text"
      value="30"
    />
    <br />
    USD
    <input
      name="usd"
      type="text"
      value="01"
    />
  </div>
</div>
`;

exports[`should 30 twd change 1 usd 1`] = `
<div>
  <div>
    <h1>
      HW10Currency
    </h1>
    <h1>
      貨幣轉換
    </h1>
    TWD
    <input
      name="twd"
      type="text"
      value="030"
    />
    <br />
    USD
    <input
      name="usd"
      type="text"
      value="1"
    />
  </div>
</div>
`;

exports[`should HW10Currency 1`] = `
<div>
  <div>
    <h1>
      HW10Currency
    </h1>
    <h1>
      貨幣轉換
    </h1>
    TWD
    <input
      name="twd"
      type="text"
      value="0"
    />
    <br />
    USD
    <input
      name="usd"
      type="text"
      value="0"
    />
  </div>
</div>
`;
